kolibri/core/assets/src/api-resource.js

```diff
@@ -237,17 +249,16 @@ class Collection {
         } else {
           this.synced = false;
           this.resource.client({ path: this.url, params }).then((response) => {
-            // Reset current models to only include ones from this fetch.
-            this.models = [];
```

Comment:
I realized this behaviour was happening, even if the returned data was malformed, so I abstracted it into the new clearCache method, and only called it when we actually had some data to set.